xchange-upbit/src/main/java/org/knowm/xchange/upbit/UpbitAdapters.java

```diff
@@ -121,7 +127,7 @@ public static Wallet adaptWallet(UpbitBalances wallets) {
   }
 
   public static Order adaptOrderInfo(UpbitOrderResponse upbitOrderResponse) {
-    Order.OrderStatus status = Order.OrderStatus.NEW;
+    Order.OrderStatus status;
     if (upbitOrderResponse.getState().equalsIgnoreCase("cancel")) {
       status = Order.OrderStatus.CANCELED;
     } else if (upbitOrderResponse.getExecutedVolume().compareTo(BigDecimal.ZERO) == 0) {
```

Comment:
the compare to == 0 implies executed volume == 0 which would be unfilled, wouldn't partially filled be a number > 0?

Reply:
Does unfilled mean that the executed value can be null?
In the demo tests, there was no case where the executed value was null.
It was 0.0 when the transaction was not executed.

Reply:
Unfilled means executed value == 0, but in this case you are saying if executedVolume == 0 that it's partially filled, which I don't think is right?

   ` } else if (upbitOrderResponse.getExecutedVolume().compareTo(BigDecimal.ZERO) == 0) {
      status = Order.OrderStatus.PARTIALLY_FILLED;`

I don't think executed volume of 0 is partially filled?

Reply:
Didn't you see the folded code?

Full code

```java
     } else if (upbitOrderResponse.getExecutedVolume().compareTo(BigDecimal.ZERO) == 0) {
       status = Order.OrderStatus.NEW;
     } else if (upbitOrderResponse.getRemainingVolume().compareTo(BigDecimal.ZERO)> 0) {
       status = Order.OrderStatus.PARTIALLY_FILLED;
```

https://github.com/knowm/XChange/pull/3811/files/3ad817e7ccd567c372ab8670af978d9b44f11214#diff-a07acc5ce0d340e36689164bcc4d69b8382ff6f6b174210b9d6a11bdcdd977c1R133-L131

Reply:
Ah, that's what I was missing, thanks for clarifying.